server/routes/basicDetails.ts

```diff
@@ -0,0 +1,411 @@
+import { type RequestHandler, Response, Router } from 'express'
+import { LocalDate, LocalDateTime } from '@js-joda/core'
+import asyncMiddleware from '../middleware/asyncMiddleware'
+import type { Services } from '../services'
+import AuditService, { Page } from '../services/auditService'
+import BreachNoticeApiClient, {
+  Address,
+  AddressList,
+  BasicDetails,
+  BreachNotice,
+  ErrorMessages,
+  Name,
+  RadioButton,
+  SelectItem,
+} from '../data/breachNoticeApiClient'
+import { fromUserDate, toUserDate } from '../utils/dateUtils'
+import { HmppsAuthClient } from '../data'
+
+export default function basicDetailsRoutes(
+  router: Router,
+  auditService: AuditService,
+  hmppsAuthClient: HmppsAuthClient,
+): Router {
+  router.get('/basic-details/:id', async (req, res, next) => {
+    await auditService.logPageView(Page.BASIC_DETAILS, { who: res.locals.user.username, correlationId: req.id })
+    const breachNoticeApiClient = new BreachNoticeApiClient(
+      await hmppsAuthClient.getSystemClientToken(res.locals.user.username),
+    )
+    const { id } = req.params
+    const basicDetails = createDummyBasicDetails()
+    const breachNotice = await breachNoticeApiClient.getBreachNoticeById(id)
+    checkBreachNoticeAndApplyDefaults(breachNotice, basicDetails)
+    const alternateAddressOptions = initiateAlternateAddressSelectItemList(basicDetails, breachNotice)
+    const replyAddressOptions = initiateReplyAddressSelectItemList(basicDetails, breachNotice)
+    const defaultOffenderAddress: Address = findDefaultAddressInAddressList(basicDetails.addresses)
+    const defaultReplyAddress: Address = findDefaultAddressInAddressList(basicDetails.replyAddresses)
+    const basicDetailsDateOfLetter: string = toUserDate(breachNotice.dateOfLetter)
+    res.render('pages/basic-details', {
+      breachNotice,
+      basicDetails,
+      alternateAddressOptions,
+      replyAddressOptions,
+      defaultOffenderAddress,
+      defaultReplyAddress,
+      basicDetailsDateOfLetter,
+    })
+  })
+
+  router.post('/basic-details/:id', async (req, res, next) => {
+    const breachNoticeApiClient = new BreachNoticeApiClient(
+      await hmppsAuthClient.getSystemClientToken(res.locals.user.username),
+    )
+    const breachNoticeId = req.params.id
+    const basicDetails: BasicDetails = createDummyBasicDetails()
+    let breachNotice: BreachNotice = null
+    breachNotice = await breachNoticeApiClient.getBreachNoticeById(breachNoticeId as string)
```

Comment:
Minor thing - don't think you need let here, you could use const instead.  Same for the other routes.
```suggestion
    const { id } = req.params
    const basicDetails = createDummyBasicDetails()
    const breachNotice = await breachNoticeApiClient.getBreachNoticeById(id)
```

Reply:
Also, if you change the `applyDefaults` function as mentioned in the other comment below - you could then just wrap the call here to simplify things e.g.
```suggestion
    const { id } = req.params
    const basicDetails = createDummyBasicDetails()
    const breachNotice = applyDefaults(await breachNoticeApiClient.getBreachNoticeById(id))
```